SolutionsAndTests

```diff
@@ -0,0 +1,258 @@
+function problem1(numbers)
+{
+	// safety checks, test for param null or not an array
+	if(	numbers == null)
+	{
+		return 0;
+	}
+	if(!(Object.prototype.toString.call( numbers ) === '[object Array]' ))
+	{
+		return 0;
+	}
+	// loop every element
+	var sum = 0;
+	for(i = 0; i < numbers.length; i++)
+	{
+		if(numbers[i] % 2 == 1 || numbers[i] % 2 == -1)
+		{
+			// consider even numbers only
+			sum += numbers[i];
+		}
+	}
+	return sum;
+}
+
+function problem2(numbers)
+{
+	// safety checks, test for param null or not an array
+	if(	numbers == null)
+	{
+		alert("problem1 ends-zero/null");
+		return 0;
+	}
+	if(!(Object.prototype.toString.call( numbers ) === '[object Array]' ))
+	{
+		alert("problem1 ends-zero/noarray");
+		return 0;
+	}
+	// init result array, [0]=biggest, [1]=2nd biggest
+	var result = new Array();
+	if(numbers.length >= 1)
+	{
+		result[0] = numbers[0];
+		if(numbers.length >= 2)
+		{
+			result[1] = Number.MIN_SAFE_INTEGER;
+		}
+	}
+	// loop every element
+	for(i = 1; i < numbers.length; i++)
+	{
+		if(numbers[i] > result[1])
+		{
+			// bigger than 2nd last biggest found
+			if(numbers[i] > result[0])
+			{
+				// bigger also than last biggest found, save previous biggest (as 2nd biggest), save current item (as new biggest)
+				result[1] = result[0];
+				result[0] = numbers[i];
+			}
+			else
+			{
+				// save over 2nd biggest
+				result[1] = numbers[i];
+			}
+		}
+	}
+	return result;
+}
+
+function problem3(numbers)
+{
+	// safety checks, test for param null or not an array
+	if(	numbers == null)
+	{
+		alert("problem1 ends-zero/null");
+		return 0;
+	}
+	if(!(Object.prototype.toString.call( numbers ) === '[object Array]' ))
+	{
+		alert("problem1 ends-zero/noarray");
+		return 0;
+	}
+	// init result
+	var result = false;
+	var n = numbers.length;
+	// loop every element
+	for(i = 0; i < n; i++)
+	{
+		var iteratingValue = numbers[i]; // take the value
+		// if the value was previously altered, obtain original value
+		var consideredValue = iteratingValue;
+		if(iteratingValue > n)
+		{
+			consideredValue = iteratingValue % n
```

Comment:
very good observation, keeping this in mind will let the code form more readable